src/resources/assets/js/store/locale.js

```diff
@@ -15,7 +15,19 @@ export const getters = {
         }
 
         const { lang } = rootState.user.preferences.global;
-        return state.i18n[lang] ? (state.i18n[lang][key] || key) : key;
+        const { env } = rootState.meta;
+        if (state.i18n[lang] && state.i18n[lang][key] !== undefined) {
+            return (state.i18n[lang][key] || key);
+        } else {
+            if (env === 'local') { 
+                axios.patch("/api/system/localisation/addLangKey", { langKey: key }).then((response) => {
+                    // not needed
+                }).catch((error) => {
+                    // not needed
+                });
```

Comment:
It's really an anti-pattern for vuex to make ajax calls from a getter. The getters should be responsible only with computing the state. To keep the convenience, we will go with this for now, but maybe in the future we can think of a better solution. Right now I can't find any other suggestion. 

Let's do

```
if (env === 'local') { 
   axios.patch("/api/system/localisation/addLangKey", { langKey: key });
}

return key;
```